include/mbgl/map/map_observer.hpp

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+namespace mbgl {
+
+class MapObserver {
+public:
+    static MapObserver& nullObserver() {
+        static MapObserver mapObserver;
+        return mapObserver;
+    }
+
+    virtual void onRegionWillChange() {}
+    virtual void onRegionWillChangeAnimated() {}
```

Comment:
Would it make sense to provide a boolean (or two-valued-enum) parameter to `on*Change` , rather than having separate `on*Change` and `on*ChangeAnimated` methods? Similarly with `on*Rendering{Frame,Map}`/`on*Rendering{Frame,Map}FullyRendered`.